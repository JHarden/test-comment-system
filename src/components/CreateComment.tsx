import React from "react";
import styled from "styled-components";

export interface CommentProps {
    onShowComment: (text: string, date: string) => void;
}

const commentStyle = {
    color: 'green',
    outline: '1px solid green'
}


const StyledComment = styled.div`
    background: #dedcdc;
    min-height: 80px;
    display: flex;
    padding: 0 0 0 20px;
    `


const Avatar = styled.div`
    width: 50px;
    height: 50px;
    background: grey;
    border-radius: 50%;
`;

const CommentBody = styled.div`
    display: flex;
    flex-direction: row;
    padding-left: 20px;
    align-items: flex-start;
    width: 80%;
    text-align: left;
    
`;

const CommentButton = styled.button`
    font-size: 20px;
    padding: 5px 0;
    height: 50px;
    width: 100px;
    background: #4ca4ff;
    position: relative;
    left: 20px;
    border-radius: 5px;
    color: #fff;
`;

const CommentInput = styled.input`
    font-size: 14px;
    min-width: 50%;
    height: 50px;
    border-radius: 5px;
    border: none;
    padding: 0 10px;
`;
class CreateComment extends React.Component<CommentProps, {}> {

    private text: string = '';
    private onSubmit = () => {
        this.props.onShowComment(this.text, new Date().toString())
    }

    private handleChange = (event: React.FormEvent<HTMLInputElement>) => {
        this.text = event.currentTarget.value;
    }

    render() {

        return (
            <StyledComment>
                <CommentBody>
                    <CommentInput type='text' placeholder={'write comment...'} onChange={this.handleChange}></CommentInput>
                    <CommentButton onClick={this.onSubmit}>send</CommentButton>
                </CommentBody>
            </StyledComment>
        )
    }
}

export default CreateComment;