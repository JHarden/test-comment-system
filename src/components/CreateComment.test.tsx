import React from 'react';
import CreateComment from './CreateComment';
import renderer from 'react-test-renderer'
const mock = () => { }

const props = {
  author: 'john',
  text: 'this is a comment!',
  date: 'yesterday',
  onShowComment: () => {}
}

test('Create Comment component test', () => {
  const component = renderer.create(
    <CreateComment {...props}></CreateComment>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
