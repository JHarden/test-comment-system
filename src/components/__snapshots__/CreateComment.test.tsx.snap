// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Comment component test 1`] = `
<div
  className="sc-bdVaJa hFJNoX"
>
  <div
    className="sc-bwzfXH bAxVqF"
  >
    <input
      className="sc-bxivhb ZaSla"
      onChange={[Function]}
      placeholder="write comment..."
      type="text"
    />
    <button
      className="sc-htpNat fNeiqx"
      onClick={[Function]}
    >
      send
    </button>
  </div>
</div>
`;
